### ISSUE TITLE
Compiling code on Windows 7

### ISSUE THREAD

**Issue by Tuttu** [ThomasHabets/simple-tpm-pk11#40]
Hey there !

I wonder if it would be possible to compile and use this code on a Windows 7 machine or if you are limited to using Unix due to some dependencies.

If possible, what would be the different requirements ?

**Comment by ThomasHabets** [comment]
It may be possible. Arping has compiled on Windows at one point, using winpcap and libnetNT. I don't know if the latter library still works.

I also have heard that Windows no longer supports raw sockets, which could be a problem.

I'd welcome pull requests and would maintain it, but not being a Windows coder I won't be actively try to make it happen.

**Comment by Tuttu** [comment]
Thanks for the quick update on the subject. Not being a Windows coder myself, I won't be able to help (Apart from testing) but I will be on the lookout for updates on the topic. :)

**Comment by ThomasHabets** [comment]
-> https://github.com/ThomasHabets/arping/issues/10

**Comment by ThomasHabets** [comment]
Sorry, I confused which project this was for. I guess the caffeine hasn't kicked in.

For simple-tpm-pk11 I have no idea where to start on Windows.

**Comment by ThomasHabets** [comment]
I guess there's also the question of what exactly do you want to do? Just use SSH with TPM?

That you can do like this: https://blog.habets.se/2016/10/Windows-SSH-client-with-TPM.html

**Comment by Tuttu** [comment]
I have more than 2000 computers using Bitlocker (Keys generated and stored in the TPM) to test and see if their TPM is affected. They _should_ be if I refer myself to all the documentation available on the Web but having the possibility to extract and test the SRK on all the TPM to be sure would be great. 

So it's not something I absolutely need to try but that would be a good thing to do nonetheless. :)

**Comment by ThomasHabets** [comment]
Oh you meant specifically the `check-srk` tool?

Do you have the tooling to start a linux live environment to run this tool? If not, it shouldn't actually be that hard to port. There seems to be [trousers for windows](https://sourceforge.net/projects/trouserswin/), so assuming it works it shouldn't take much time to make check-srk build on Windows.

**Comment by Tuttu** [comment]
Yep. Sorry, I figured too late the check-srk tool was just a part of the simple-tpm-pk11.

As for your question, the answer is no. We can't deploy a live Linux environment on the workstations as the process must be completely silent to the user. I guess I will just wait and see if a similar windows tool pops on the Web. :)

**Comment by ThomasHabets** [comment]
I've asked on [trousers-users](https://sourceforge.net/p/trousers/mailman/message/36092876/)

**Comment by Tuttu** [comment]
Oh, thanks for the help. 👍

**Comment by ThomasHabets** [comment]
What you could do is generate a new key as described [here](https://blog.habets.se/2016/10/Windows-SSH-client-with-TPM.html) and then use the official scripts to test the key.

* https://github.com/crocs-muni/roca
* https://keychest.net/roca ("offline test" tab)

**Comment by Tuttu** [comment]
That's indeed a good idea. I will see what I can do from a test computer.

**Comment by hajikhorasani** [comment]
@Tuttu and @ThomasHabets  I couldn't find the Windows version of TPM-PKCS11. I'll be appreciated if you guide me. I want to use a eToken Pro PKCS11 and TPM-PKCS11 simultaneously in a Windows machine by CryptokiMPX
https://github.com/hajikhorasani/cryptokimpx

**Comment by ThomasHabets** [comment]
@hajikhorasani like I said for coding on Windows I don't even know where to start. And nobody replied to the thread I started (linked above). I've not coded for Windows in about 20 years.
